src/main/java/com/nutomic/syncthingandroid/syncthing/EventProcessor.java

```diff
@@ -0,0 +1,171 @@
+package com.nutomic.syncthingandroid.syncthing;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.os.Environment;
+import android.os.Handler;
+import android.os.Looper;
+import android.preference.PreferenceManager;
+import android.support.v4.content.LocalBroadcastManager;
+import android.util.Log;
+import android.widget.Toast;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import eu.chainfire.libsuperuser.Shell;
+
+/**
+ * Run by the syncthing service to convert syncthing events into local broadcasts.
+ * It uses SyncthingService.GetEvents to read the pending events and wait for new events.
+ */
+public class EventProcessor implements SyncthingService.OnWebGuiAvailableListener, Runnable, RestApi.OnReceiveEventListener {
+
+    private static final String TAG = "EventProcessor";
+    private static final String PREF_LASTSYNCID = "last_sync_id";
+
+    private static final String EVENT_BASE_URL = "com.nutomic.syncthingandroid.event";
+
+    /**
+     * Minimum interval in seconds at which the events are polled from syncthing and processed.
+     * This intervall will not wake up the device to save battery power.
+     */
+    public static final long EVENT_UPDATE_INTERVAL = 15;
```

Comment:
I meant that you should put the `TimeUnit.SECONDS.toMillis(15)` directly here, so you don't have to convert it later. Because this way, it's not clear what unit is used.